core/electron/src/backend/ElectronHost.ts

```diff
@@ -278,11 +278,20 @@ export class ElectronHost {
 class ElectronAppHandler extends IpcHandler {
   public get channelName() { return "electron-safe"; }
   public async callElectron(member: string, method: string, ...args: any) {
-    const electronMember = (ElectronHost.electron as any)[member];
-    const func = electronMember[method];
-    if (typeof func !== "function")
-      throw new IModelError(IModelStatus.FunctionNotFound, `Method ${method} not found electron.${member}`);
+    let allowedMethods: readonly string[] = [];
+    if (member === "dialog") {
+      const methods: readonly (keyof Electron.Dialog)[] = ["showMessageBox", "showOpenDialog", "showSaveDialog"];

```

Comment:
Any way we could set the type in a common file that way the backend and frontend could share the list?

Reply:
Yes, that's a good idea :)

Reply:
@calebmshafer I've renamed the type `DialogModuleMethods` to singular: `DialogModuleMethod` and moved it to `common` directory. 

Alternatively, we could create a tuple w/ const assertion and get the union type from it, i.e.:
```
function dialogModuleMethod<T extends AsyncMethodsOf<Electron.Dialog>>(method: T) {
  return method;
}

/** @internal */
export const dialogModuleMethods = [
  dialogModuleMethod("showMessageBox"),
  dialogModuleMethod("showOpenDialog"),
  dialogModuleMethod("showSaveDialog"),
] as const;

/** Exposed main process methods of dialog module in an Electron app.
 * @beta
 */
export type DialogModuleMethod = typeof dialogModuleMethods[number];
```

We could then use this `dialogModuleMethods` directly in the handler as a bonus point, however it would require us to mark it as @beta as well, that's why I've avoided doing that (for documentation/extract-api).

Reply:
Yea I agree that keeping it internal for now is probably the better approach. Then if we see the need we can change it later.